### ISSUE TITLE
Code Coverage: babel-generator

### ISSUE THREAD

**Issue by hzoo** [babel/babel#5326]
### Context

When I was reviewing https://github.com/babel/babel/pull/5317 again I noticed we lost some code coverage in the PR.

![screen shot 2017-02-16 at 11 49 36 am](https://cloud.githubusercontent.com/assets/588473/23031166/021ba892-f43e-11e6-9b1b-f2659b6224bc.png)

^ The red boxes are the places where coverage was decreased. I clicked on the red box to get

![screen shot 2017-02-16 at 11 49 52 am](https://cloud.githubusercontent.com/assets/588473/23031195/1ac484ea-f43e-11e6-8b42-e027c1e8a372.png)

This shows the code where we don't have a test to cover that logic.

For `babel-generator`, it just means we have to add a few tests! In the case of this particular package, it's as simple as writing out the relevant code snippets.

---

> You can ask questions in our [Slack](http://slack.babeljs.io/) or below

- [ ] Comment below you are going to do this (for others to know) - for this particular issue we can have multiple people working on this. I would just recommend that you do a multiple small PRs instead of trying to make the coverage 100% in one PR or by yourself
- [ ] Read [CONTRIBUTING.md](https://github.com/babel/babel/blob/master/CONTRIBUTING.md)
- [ ] remember to run `make build` after a source code change or better just run `make watch` in another terminal window to auto run babel while you are developing 

- [ ] Go to https://codecov.io/gh/babel/babel/tree/master/packages/babel-generator/src and click on a folder/file that has less than 100% coverage in `generators/` or `node/`

Example in https://codecov.io/gh/babel/babel/src/master/packages/babel-generator/src/node/whitespace.js

![screen shot 2017-02-16 at 11 52 57 am](https://cloud.githubusercontent.com/assets/588473/23031297/7c6d23d2-f43e-11e6-9d2d-081adc52effb.png)

Reading that it looks like we just need to add some code that is a `LogicalExpression` (`a || b`) where the `node.left` is a function.

> If you don't understand what an AST Node is (like `LogicalExpression`), then I would use a combination of [ASTExplorer](http://astexplorer.net/#/gist/1ce0ec276cfaa115a2ea6e8349725a86/0c16d93a489b1e50ed75f0a78e254d6ae2861a09) and our [spec docs](https://github.com/babel/babylon/blob/master/ast/spec.md). I believe JSX/Flow isn't documented completely so you may need to reference the [JSX docs](https://facebook.github.io/jsx/) and [Flow docs](https://flowtype.org/docs/getting-started.html#_). And actually that should be it's own issue.

- [ ] add a test(s) in the test folder: https://github.com/babel/babel/tree/master/packages/babel-generator/test/fixtures based on the category or in the `types` folder https://github.com/babel/babel/tree/master/packages/babel-generator/test/fixtures/types.

If there is no folder then feel free to make a new one or ask.

Example with ArrowFunctionExpression: https://github.com/babel/babel/tree/master/packages/babel-generator/test/fixtures/types/ArrowFunctionExpression

A test is just the `actual.js` (input code) and `expected.js` (output code). They should usually be the exact same unless the test is dealing with something like whitespace, parentheses, or different formatting of something like numbers.

```js
// in this case, it's the same
var foo = x => x * x;
var foo = (a, b) => a * b;
var foo = async x => x * x;
var foo = async (a, b) => a * b;
```

**Comment by hzoo** [comment]
> If you need any help, or just have general Babel or JavaScript questions, we have a vibrant Slack
community that typically always has someone willing to help. You can sign-up [here](https://slack.babeljs.io/)
for an invite.

**Comment by alxpy** [comment]
@hzoo I'd like to work on this, if you don't mind

**Comment by aaronang** [comment]
@hzoo I would also like to help out with writing tests. I am assuming that ideally we would like to achieve 100% test coverage?

**Comment by keshcode** [comment]
i would like to work on this

**Comment by Thuva4** [comment]
I would like to work on this issue

**Comment by Hoten** [comment]
Looking to contrib, this looks like a good issue to start with. I'll proceed by making a separate PR for each test I add.

**Comment by Hoten** [comment]
Having trouble getting the test coverage to work locally.

I ran this first: `BABEL_ENV=cov make build`

Then,

`sudo ./scripts/test-cov.sh`
---
```
(snip)
1848 passing (50s)
  7 pending

----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
All files |  Unknown |  Unknown |  Unknown |  Unknown |                |
----------|----------|----------|----------|----------|----------------|
```

`cat .nyc_output/*`
---
```
{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}
```

Nothing in the output folder. `coverage/coverage-final.json` is also empty.

What am I missing?

**Comment by 1egoman** [comment]
@Hoten I had luck first running `make test-ci-coverage` to let the system initialize itself, then running `./scripts/test-cov.sh  && node_modules/.bin/nyc report --reporter html` to update coverage results when I made a change. I bet there's a better way to do this, though.

**Comment by nveenjain** [comment]
I would like to work on this issue as my first PR.

**Comment by dantesolis** [comment]
Hey can I still work on this?

**Comment by nicolo-ribaudo** [comment]
Hi @dantesolis, I think this issue has already been addressed by #5934

**Comment by debugpai2** [comment]
Can this be closed now if it is no longer an issue?
